### DISCUSSION TITLE
Confusion regarding Reach_Ros plugins

### DISCUSSION THREAD

**Discussion by natalieCloud** [ros-industrial/reach#60] [Q&A] [answered]
Hello!

So, currently I'm experiencing an error that I believe is somewhat similar to those mentioned [here](https://github.com/ros-industrial/reach_ros2/issues/13#issuecomment-1605261794):

```
Failed to create plugin instance 'MoveItIKSolver' of type 'reach::IKSolverFactory'
Search Paths (including system folders)
Search Libraries:
    - libreach_plugins.so
Available plugins of type 'reach::IKSolverFactory':
    - NoOpIKSolver
```

At first I believed it to be a cmake/inclusion error, (as it was originally produced through running a node similar to the base reach_study_node in a seperate package), however when attempting to run the original reach_study_node through the base package I got the same result. What is especially interesting is that when I run the node through the provided launch file (& the same configurations) I get no such error. 

Was wondering if there were any thoughts on this or ideas as to if I could be misinterpreting the build?

Additional info: Using colcon & ROS2 Humble 

Thank you for your help!

**Comment by marip8** [comment, marked as answer]
> What is especially interesting is that when I run the node through the provided launch file (& the same configurations) I get no such error.

Did you set the `REACH_PLUGINS` environment variable? The nominal demo launch file adds the name of the `reach_ros` plugin library (i.e., `reach_ros_plugins`) to this environment variable [here](https://github.com/ros-industrial/reach_ros2/blob/d3ac7bba99f8a4e5d0e9cd550bca3c94967ca73c/launch/start.launch.py#L58). See [this section of the README of this repo](https://github.com/ros-industrial/reach#plugins) for more details about how to use plugins built outside of this repo

**Reply by natalieCloud** [reply]
It worked great! Thank you so much!!
